docs/architecture/decisions/0004-choose-http-client-library.md

```diff
@@ -0,0 +1,31 @@
+# 1. Choose the best HTTP Client library on Node JS
```

Comment:
A couple of warnings:

1. since the http(s) calls that a function can make in parallel are limited, in the whole IO codebase we rely on a [specific implementation of Http(s) agent](https://github.com/pagopa/io-ts-commons/blob/master/src/agent.ts) that uses node-fetch to set up the right **keepalive** settings taking values from the environment; without this setup the function app will starve. if you want to use an alternative you need to ensure that the correct agent is passed on boostrap (backporting that implementation)
1. you need to ensure that your fetch calls have a **timeout**; in order to do that we use a custom implementation of an Abortable fetch that uses the Abort agent, see for example here: https://github.com/pagopa/pagopa-checkout-fe/blob/main/src/utils/config/fetch.ts#L35

cc @lucacavallaro @grausof

Reply:
Thank you! In the final implementation I will take these warnings into consideration!